internal/pkg/template/input/validation.go

```diff
@@ -73,18 +95,55 @@ func validateInputIf(fl goValidator.FieldLevel) bool {
 func validateUserInputTypeByKind(value interface{}, kind string) error {
 	inputType := reflect.TypeOf(value).String()
 	switch kind {
-	case "password", "textarea":
+	case KindPassword, KindTextarea:
 		if inputType != reflect.String.String() {
 			return fmt.Errorf("the input is of %s kind and should be a string, got %s instead", kind, inputType)
 		}
-	case "confirm":
+	case KindConfirm:
 		if inputType != reflect.Bool.String() {
 			return fmt.Errorf("the input is of confirm kind and should be a bool, got %s instead", inputType)
 		}
 	}
 	return nil
 }
 
+func validateUserInputByType(userInput interface{}, inputType string) error {
+	switch inputType {
+	case reflect.Float64.String():
+		return validateUserInputByReflectKind(userInput, reflect.Float64)
+	case reflect.Int.String():
+		return validateUserInputByReflectKind(userInput, reflect.Int)
+	case reflect.String.String():
+		return validateUserInputByReflectKind(userInput, reflect.String)
+	default:
+		panic(fmt.Errorf("unknown type %s", inputType))
+	}
+}
+
+func validateUserInputByReflectKind(userInput interface{}, expectedType reflect.Kind) error {
+	userInputIsValue := reflect.TypeOf(userInput).String() == "reflect.Value"
```

Comment:
Tady jsem se trochu zapotil, protože když se sem přijde z validace definice z `validateInputDefault` tak je v `userInput` reflect.Value, kterou ten Validator všechno obaluje. Teď teda ještě přemýšlím jestli by nakonec nebylo lepší ty dva use casy úplně oddělit (tj. zvlášť validace typu pro user input a zvlášť pro definici defaultu), ale to už nechám na tobě. 😅